sopt/src/main/scala/dagr/sopt/util/ParsingUtil.scala

```diff
@@ -0,0 +1,141 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.sopt.util
+
+import java.lang.reflect.Modifier
+
+import dagr.commons.reflect.ReflectionUtil
+import dagr.commons.util.ClassFinder
+import dagr.sopt.cmdline._
+import dagr.sopt.clp
+
+import scala.collection.JavaConversions._
+import scala.collection.immutable.Map
+import scala.reflect._
+import scala.reflect.runtime.universe._
+
+/** Variables and Methods to support command line parsing */
+object ParsingUtil extends Configuration  {
+  /** Gets the [[clp]] annotation on this class */
+  def findClpAnnotation(clazz: Class[_]): Option[clp] = {
+    ReflectionUtil.findJavaAnnotation(clazz, classOf[clp])
+  }
+
+  /** True if we are to use ANSI colors to print to the terminal, false otherwise. */
+  val printColor: Boolean = optionallyConfigure[Boolean](Configuration.Keys.ColorStatus).getOrElse(true)
+
+  /** Initializes the color for printing */
+  private def initializeColor(color: String): String = {
+    if (printColor) color
+    else ""
+  }
+
+  /** Provides ANSI colors for the terminal output **/
+  val KNRM: String = initializeColor("\u001B[0m")
+  //val KBLD: String = initializeColor("\u001B[1m")
+  val KRED: String = initializeColor("\u001B[31m")
+  val KGRN: String = initializeColor("\u001B[32m")
+  val KYEL: String = initializeColor("\u001B[33m")
+  val KBLU: String = initializeColor("\u001B[34m")
+  val KMAG: String = initializeColor("\u001B[35m")
+  val KCYN: String = initializeColor("\u001B[36m")
+  val KWHT: String = initializeColor("\u001B[37m")
+  val KBLDRED: String = initializeColor("\u001B[1m\u001B[31m")
+  val KERROR: String = initializeColor("\u001B[1m\u001B[31m")
+  //val KERROR: String = initializeColor("\u001B[35;7;1m") // BLINKING: "\u001B[35;5;1m"
+
+  /** Gets a mapping between sub-classes of [[T]] and their [[clp]] annotation.
+    *
+    * @param srcClasses the source classes from which to build the map.
+    * @param omitSubClassesOf a list of omit subclasses to omit from the returned map.
+    * @param includeHidden include classes whose [[clp]] annotation has `hidden` set to true.
+    * @tparam T the super class.
+    * @return a mapping between sub-classes of [[T]] and their [[clp]] annotation.
+    */
+  private def classToPropertyMapFromSourceClasses[T](srcClasses: Traversable[Class[_]],
+                                                     omitSubClassesOf: Iterable[Class[_]] = Nil,
+                                                     includeHidden: Boolean = false)
+  : Map[Class[_ <: T], clp] = {
+
+    // Filter out interfaces, synthetic, primitive, local, or abstract classes.
+    def keepClass(clazz: Class[_]): Boolean = {
+      !clazz.isInterface && !clazz.isSynthetic && !clazz.isPrimitive && !clazz.isLocalClass && !Modifier.isAbstract(clazz.getModifiers)
```

Comment:
As it stands, that won't work unless we do some hackery to override the access modifier on the class and/or constructor.  We'll get a security violation exception when we try to instantiate it.